src/Selenium2Library/locators/elementfinder.py

```diff
@@ -151,6 +151,17 @@ def _find_by_default(self, browser, criteria, tag, constraints):
         )
         return self._normalize_result(browser.find_elements_by_xpath(xpath))
 
+    def _get_xpath_constraints(self, constraints):
+        xpath_constraints = []
+        for name in constraints:
+            value = constraints[name]
+            if isinstance(value, list):
+                contraint = "@%s[. = '%s']" % (name, "' or . = '".join(value))
+            else:
+                contraint = "@%s='%s'" % (name, value)
+            xpath_constraints.append(contraint)
+        return xpath_constraints
```

Comment:
Made this change too. It's much better. I was just afraid of creating a new method because all other methods in the file are relatively long. I really like having smaller and more methods. ^^